src/frontend/type_checker.cpp

```diff
@@ -173,6 +173,30 @@ void TypeChecker::visit(SetType::Ptr type) {
   }
 }
 
+void TypeChecker::visit(LatticeLinkSetType::Ptr type) {
+  retTypeChecked = typeCheck(type->element);
+  
+  retTypeChecked = typeCheck(type->latticePointSet) && retTypeChecked;
+  // Check lattice point set is an unstructured set
+  if (!isa<UnstructuredSetType>(type->latticePointSet->set->setDef)) {
+    std::stringstream errMsg;
+    errMsg << "expected lattice point set of Unstructured kind but got "
+           << type->latticePointSet->set->setName;
+    reportError(errMsg.str(), type->latticePointSet);
+    retTypeChecked = false;
```

Comment:
`reportError` actually sets `retTypeChecked` to false automatically, so don't need to do it again here.